### ISSUE TITLE
max_age=0 sets skip == true when it should be false

### ISSUE THREAD

**Issue by tbalasavage** [ory/hydra#3034]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/hydra/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/hydra/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

When an authentication request has a `max_age` value of 0 (`max_age=0`), Hydra sets `skip` == `true` for a remembered authentication session. I would instead expect that `skip` == `false` and a login should be forced. As far as I can see, 0 is a valid value for `max_age`. According to the OpenID spec, `max_age` corresponds to OpenID 2.0 [PAPE](https://openid.net/specs/openid-connect-core-1_0.html#OpenID.PAPE) [OpenID.PAPE] max_auth_age request parameter, which states the following:
> Value: Integer value greater than or equal to zero in seconds.  

I think that the issue is in the code here where a valid `max_age` must be greater than 0:
```go
if maxAge > 0 && time.Time(session.AuthenticatedAt).UTC().Add(time.Second*time.Duration(maxAge)).Before(time.Now().UTC()) {
if stringslice.Has(prompt, "none") {
	return errorsx.WithStack(fosite.ErrLoginRequired.WithHint("Request failed because prompt is set to 'none' and authentication time reached 'max_age'."))
}
return s.forwardAuthenticationRequest(w, r, ar, "", time.Time{}, nil)
}
```

Auth0 has a [nice article](https://auth0.com/docs/authenticate/login/max-age-reauthentication) about forcing authentication in OIDC and their recommendation within the article is to set `max_age=0`. 

I'd be willing to submit a PR to update this but would like to know whether or not such a change would be accepted. 

### Reproducing the bug

Submit an authentication request with `max_age=0` for a previously remembered authentication session. Assuming a properly configured login service, the login screen is not shown to the user. 

### Version

1.10

**Comment by aeneasr** [comment]
Without having done a lot of introspection, I think you're right here. I guess the problem is that `0` is the default value for ints, but here `0` being set has actually a different meaning than when it's not set (null).

At the same time though, we are passing all OIDC certification tests which include tests for `max_age=0` so I'm not 100% what the issue is here and why it wasn't caught by OIDC certification

**Comment by tbalasavage** [comment]
I'm not sure either. Can I help in some way? I've remedied this within the login service by checking for a `max_age=0` but it would be nice if we didn't need to rely on this.

**Comment by aeneasr** [comment]
It looks like this is a special case that has been later added to the spec, according to the linked article:

```
max_age=0: Effectively force a login prompt using only the max_age parameter. Note that a recent spec update further clarified this parameter,[ stating it is effectively the same as ](https://bitbucket.org/openid/connect/commits/90199f36a1f8)prompt=login[.](https://bitbucket.org/openid/connect/commits/90199f36a1f8) This one is not feasible since it blends what should be a UX parameter with a session maintenance parameter.
```

Thus, I would say that a workaround is to use `prompt=login` instead of `max_age=0` for now. Fixing `max_age=0` is still on the table though!

**Comment by aeneasr** [comment]
Looks like my second comment didn't get through, I found this doc:

https://openid.net/specs/openid-connect-basic-1_0.html

which says

```
max_age
OPTIONAL. Maximum Authentication Age. Specifies the allowable elapsed time in seconds since the last time the End-User was actively authenticated by the OP. If the elapsed time is greater than this value, the OP MUST attempt to actively re-authenticate the End-User. When max_age is used, the ID Token returned MUST include an auth_time Claim Value. Note that max_age=0 is equivalent to prompt=login.
```

**Comment by tbalasavage** [comment]
If a client sends `prompt=login` but not `max_age=0`, will that effect the presence of the `auth_time` claim though? If we want to ensure it's present then, I assume then that should they include both because the spec says `auth_time` is required when `max_age` is present but doesn't mention anything related to `prompt`.

**Comment by tbalasavage** [comment]
The other thing I'm seeing is that because `max_age=0` is not honored, the login request has `subject` set to user ID from the cookie. If you try to accept a login with a different subject, the request fails. If you use `max_age=1` or `prompt=login`, `subject` is empty and you can accept a login.

**Comment by aeneasr** [comment]
Yes, I think it makes sense for us to fix this behavior. Accepting PRs :) But please make them against the v2.x branch! :)
